### PR TITLE
Feature purchase integration

### PR DESCRIPTION
**Pull request by DarrylEgbertNangoi-GitHub**
Screen Shot 2024-06-26 at 23 50 16

Screen Shot 2024-06-26 at 23 50 21

### CODE REVIEW

**Review by LeroyMichael — CHANGES_REQUESTED** [review]
image
pas ngubah items qty, totalnya ga langsung ke update

**Comment by LeroyMichael** [comment]
image
add items kurang description

**Comment by LeroyMichael** [comment]
kalau transactionnya udah active, pagenya ga bsa diedit
WhatsApp Image 2024-06-28 at 16 35 53_6326f4d6
pas submit kluar notif error ini

File: lib/contacts/utils.ts [hunk 1]
```diff
@@ -3,10 +3,14 @@ import { Contact } from "@/types/contact.d";
...
-  currentPage: Number
+  currentPage: Number,
+  search?: string
 ): Promise<Record<string, any>> {
+
+  const searchParams = search && `&search=${search}`
+  
   const res = await fetch(
-    `${process.env.NEXT_PUBLIC_URL}/api/contacts?merchant_id=${merchant_id}&page=${currentPage}`,
+    `${process.env.NEXT_PUBLIC_URL}/api/contacts?merchant_id=${merchant_id}&page=${currentPage}${searchParams}`,
```
Comment (LeroyMichael) [inline lib/contacts/utils.ts:13]: ini hrsnya
`&page=${currentPage}&search=${searchParams}`

**Review by LeroyMichael — CHANGES_REQUESTED** [review]

File: lib/contacts/utils.ts (hunk 1 above)
Comment (DarrylEgbertNangoi-GitHub) [reply lib/contacts/utils.ts:13]: searchparams kan udh di set diatasnya  si res

Comment (DarrylEgbertNangoi-GitHub) [reply lib/contacts/utils.ts:13]: Screen Shot 2024-06-30 at 15 47 51

**Pull request merged**